### PR TITLE
Fixed a bug in python object refcounts, where isType() consumed refcount for type objects like `PyBool_Type`, triggering segfaults after repeated invocation.

### PR DESCRIPTION
**Pull request by mhong**
The root cause was tracked down by instrumenting Py_IncRef and Py_DefRef calls, and printing out object refcounts. Will try to clean up and submit such debugging infra in later patches.

This fixes TF-78 (https://bugs.swift.org/browse/TF-78).

### CODE REVIEW

**Comment by mhong** [comment]
@swift-ci please test tensorflow

File: test/Python/python_runtime.swift [hunk 1]
```diff
@@ -257,4 +257,14 @@ PythonRuntimeTestSuite.test("SR-9230") {
   expectEqual(2, Python.len(Python.dict(a: "a", b: "b")))
 }
 
+// TF-78: isType() consumed refcount for type objects like `PyBool_Type`.
+PythonRuntimeTestSuite.test("PythonRefCount") {
+  let b: PythonObject = true
+  for _ in 0...20 {
+    // This triggers isType(), which used to crash after repeated invocation
+    // because of reduced refcount for `PyBool_Type`.
+    _ = Bool.init(b)
```
Comment (rxwei) [inline test/Python/python_runtime.swift]:
```suggestion
    _ = Bool(b)
```

**Review by rxwei — APPROVED** [review]
Thanks!

**Review by dan-zheng — APPROVED** [review]
Great find!

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Pull request merged**